packages/trading-widget/src/core-kit/hooks/trading/withdraw-v2/swap-step/use-withdraw-swap-quote.ts

```diff
@@ -0,0 +1,48 @@
+import BigNumber from 'bignumber.js'
+import { useEffect } from 'react'
+
+import { useReceiveTokenInput } from 'core-kit/hooks/state'
+import {
+  useWithdrawSwapData,
+  useWithdrawTrackedAssets,
+} from 'core-kit/hooks/trading/withdraw-v2/swap-step'
+import { formatUnits, isEqualAddress } from 'core-kit/utils'
+
+export const useWithdrawSwapQuote = () => {
+  const [receiveAsset, updateReceiveToken] = useReceiveTokenInput()
+  const { data: assets = [] } = useWithdrawTrackedAssets()
+  const { data: swapData, isFetching, isError } = useWithdrawSwapData()
+
+  useEffect(() => {
+    updateReceiveToken({ isLoading: isFetching })
+  }, [isFetching, updateReceiveToken])
+
+  useEffect(() => {
+    if (assets.length === 0 || isFetching || isError) {
```

Comment:
Is there a chance where `assets.length === 0` could be a valid value and should be processed?

Reply:
No, total received value can be calculated only when assets.length > 0